include/libtorrent/kademlia/dht_tracker.hpp

```diff
@@ -69,14 +71,20 @@ namespace libtorrent { namespace dht {
 			, counters& cnt
 			, dht_storage_interface& storage
 			, dht_state state);
-		virtual ~dht_tracker();
+
+		dht_tracker(dht_tracker const&) = delete;
+
+		virtual ~dht_tracker() = default;
```

Comment:
I think you can just remove this destructor declaration. This class is final, so nobody can derive from it, and if ``socket_manager`` has a virtual destructor, this destructor will be virtual automatically anyway.